mount/mount_windows.go

```diff
@@ -17,17 +17,44 @@
 package mount
 
 import (
+	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/Microsoft/go-winio/pkg/bindfilter"
 	"github.com/Microsoft/hcsshim"
+	"github.com/containerd/containerd/log"
+	"golang.org/x/sys/windows"
+)
+
+const sourceStreamName = "containerd.io-source"
+
+var (
+	// ErrNotImplementOnWindows is returned when an action is not implemented for windows
+	ErrNotImplementOnWindows = errors.New("not implemented under windows")
 )
 
 // Mount to the provided target.
 func (m *Mount) mount(target string) error {
+	readOnly := false
+	for _, option := range m.Options {
+		if option == "ro" {
+			readOnly = true
+			break
+		}
+	}
```

Comment:
We use this here and in the snapshotter, can we add a `(m *Mount) ReadOnly() bool` function?

Reply:
Frankly, "search the `mount(8)` command-line options array for 'ro'" approach has always bugged me (but not enough to propose doing things differently, given that this is an API type). _Hiding_ it would be a nice first step.

Reply:
@helsaawy Yes. Definitely. Changing now.